sdk/iOS/test/MSSyncTableTests.m

```diff
@@ -1082,9 +1083,380 @@ -(void) testPushNetworkTimeoutWithErroredOps
+        XCTAssertNil(error);
+        
+        NSString *responseString = [NSString stringWithFormat:@"{\"id\": \"%@\", \"text\":\"server\"}", item[@"id"]];
+        NSData *pushResponse = [responseString dataUsingEncoding:NSUTF8StringEncoding];
+
+        return pushResponse;
+    };
+
+    // Perform push
+    XCTestExpectation *pushExpectation = [self expectationWithDescription:@"push expectation"];
+    [filteredClient.syncContext pushWithCompletion:^(NSError *error) {
+        XCTAssertNil(error, "Expected successful push");
+        [pushExpectation fulfill];
+    }];
+    [self waitForExpectationsWithTimeout:10 handler:nil];
+    
+    return [self synchronizedItemCount];
+}
+
+// Returns the number of synchronized items, i.e. items that are present on the server as well as the client.
+// An item is synchronized if it was successfully pushed to the server from the client OR
+// if it was successfully pulled from the server to the client.
+-(int) synchronizedItemCount
+{
+    // Initialization
+    MSSyncTable *todoTable = [client syncTableWithName:TodoTableNoVersion];
+    MSQuery *query = [[MSQuery alloc] initWithSyncTable:todoTable];
+    query.predicate = [NSPredicate predicateWithFormat:@"text == 'server'"];
+
+    // Number synchronized items
+    __block int numSynchronizedItems = 0;
+    
+    XCTestExpectation *readExpectation = [self expectationWithDescription:@"read expectation"];
+    
+    // Read from the table
+    [client.syncContext readWithQuery:query completion:^(MSQueryResult *result, NSError *error) {
+        
+        numSynchronizedItems = result.items.count;
+        [readExpectation fulfill];
+        
+    }];
+    
+    [self waitForExpectationsWithTimeout:10 handler:nil];
+    
+    return numSynchronizedItems;
+}
+
+// Verify that the specified NSOperation has finished
+-(void) verifyFinished:(NSOperation *)operation
+{
+    XCTestExpectation *completionExpectation = [self expectationWithDescription:@"Completion expectation"];
+    
+    // Define an operation and make it dependent on blockingOperation
+    NSOperation *dependentOperation = [NSBlockOperation blockOperationWithBlock:^{
+        [completionExpectation fulfill];
+    }];
+
+    // Add the dependentOperation to the queue after configuring dependencies
+    [dependentOperation addDependency:operation];
+    [[NSOperationQueue new] addOperation:dependentOperation];
+    
+    [self waitForExpectationsWithTimeout:10 handler:nil];
+}
 
 #pragma mark Pull Tests
 
+// Tests cancellation of the pull operation at various stages of the pull workflow
+// - Verifies that a pull operation following a cancelled pull operation works as expected
+// - Verifies that cancelling a pull operation finishes the pull NSOperation regardless of the cancellation point
+// - Verifies that the expected number of items are pulled from the server
+-(void) testPullCancellability_successfulpush_successfulpull
```

Comment:
I would still prefer each cancel point be its own test method.  Its unclear what is gained by combining them like so. Now you are dealing with previous test state/etc that would just be avoided by breaking them all up.